lib/internal/main/watch_mode.js

```diff
@@ -103,27 +103,25 @@ async function stop() {
   clearGraceReport();
 }
 
+let restarting = false;
 async function restart() {
+  if (restarting) return;
+  restarting = true;
   if (!kPreserveOutput) process.stdout.write(clear);
   process.stdout.write(`${green}Restarting ${kCommandStr}${white}\n`);
   await stop();
   start();
+  restarting = false;
 }
 
-(async () => {
-  emitExperimentalWarning('Watch mode');
-
-  try {
-    start();
-
-    // eslint-disable-next-line no-unused-vars
-    for await (const _ of on(watcher, 'changed')) {
-      await restart();
-    }
-  } catch (error) {
+emitExperimentalWarning('Watch mode');
+start();
+watcher
+  .on('changed', restart)
```

Comment:
do you need to set `captureRejections` to true on the watcher to get the same behavior as before? IIRC you need to set `captureRejections` true to have the same behavior as before (that the error will be called if restart rejects)